XenModel/Actions/Updates/DownloadAndUnzipXenServerPatchAction.cs

```diff
@@ -60,12 +60,14 @@ public string PatchPath
             get; private set;
         }
 
-        public DownloadAndUnzipXenServerPatchAction(string patchName, Uri uri, string outputFileName)
-            : this(patchName, uri, outputFileName, InvisibleMessages.XEN_UPDATE)
-        { }
+        private bool UnzipOnly
+        {
+            get { return address == null; }
+        }
 
-        public DownloadAndUnzipXenServerPatchAction(string patchName, Uri uri, string outputFileName, params string[] updateFileExtensions)
-            : base(null, string.Format(Messages.DOWNLOAD_AND_EXTRACT_ACTION_TITLE, patchName), string.Empty, false)
+        public DownloadAndUnzipXenServerPatchAction(string patchName, Uri uri, string outputFileName, bool suppressHist, params string[] updateFileExtensions)
+            : base(null, uri == null ? string.Format(Messages.DOWNLOAD_AND_EXTRACT_ACTION_TITLE, patchName)

```

Comment:
The action title should be assigned the other way around: if uri is null, then UPDATES_WIZARD_EXTRACT_ACTION_TITLE, else DOWNLOAD_AND_EXTRACT_ACTION_TITLE